sch_cake.c

```diff
@@ -1637,28 +1665,50 @@ static u8 cake_handle_diffserv(struct sk_buff *skb, u16 wash)
 {
 	int wlen = skb_network_offset(skb);
```

Comment:
Since you're now using this as an offset rather than a length, I think it's better to rename it...